lib/rubocop/cop/lint/to_json.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks to make sure `#to_json` includes an optional hash
+      # argument.  When overriding `#to_json`, callers may invoke JSON
+      # generation via `JSON.generate(your_obj)`.  Since `JSON#generate` allows
+      # for optional arguments, your method should too.
+      #
+      # @example
+      #   # bad
+      #   def to_json
+      #   end
+      #
+      #   # good
+      #   def to_json(opts = {})
```

Comment:
As mentioned in https://github.com/rubocop-hq/rubocop/issues/6378#issuecomment-441320824 I am not sure the argument is always a Hash

```rb
require 'json'
class MyClass
  def to_json(*args)
    p args
    ""
  end
end
JSON.generate(MyClass.new)
#=> [#<JSON::Ext::Generator::State:0x00007faacb155a40>]
```

In the JSON documentation I see a few different signatures for `#to_json`:

 - `to_json(*)` [(source)](http://ruby-doc.org/stdlib-2.5.1/libdoc/json/rdoc/JSON/Ext/Generator/GeneratorMethods/FalseClass.html#method-i-to_json)
 - `to_json(*args)` [(source)](http://ruby-doc.org/stdlib-2.5.1/libdoc/json/rdoc/Date.html#method-i-to_json)
 - `to_json(state = nil)` [(source)](http://ruby-doc.org/stdlib-2.5.1/libdoc/json/rdoc/JSON/Ext/Generator/GeneratorMethods/Hash.html#method-i-to_json)

…maybe we can only say that `to_json` should have arity 1?

Reply:
I agree @bquorning - I'll make that change

Reply:
On second thought, `*` and `*args` aren’t arity 1…